src/herder/test/HerderTests.cpp

```diff
@@ -3023,6 +3023,134 @@ TEST_CASE("SCP checkpoint", "[catchup][herder]")
+
+        auto tx1 = root.tx({createAccount(a1, minBalance2)});
+        auto tx2 = root.tx({createAccount(b1, minBalance2)});
+
+        return std::make_tuple(root, a1, b1, tx1, tx2);
+    };
+
+    SECTION("mixed")
+    {
+        setup(true);
+
+        auto [root, a1, b1, tx1, tx2] = makeTxs(app);
+
+        // Submit txs for the same account, should be good
+        REQUIRE(app->getHerder().recvTransaction(tx1, true) ==
+                TransactionQueue::AddResult::ADD_STATUS_PENDING);
+        REQUIRE(app->getHerder().recvTransaction(tx2, true) ==
+                TransactionQueue::AddResult::ADD_STATUS_PENDING);
+
+        // Leader election here should be completely determenistic based on
+        // node IDs
+        uint32_t lcl = app->getLedgerManager().getLastClosedLedgerNum();
+        simulation->crankUntil(
+            [&]() {
+                return app->getLedgerManager().getLastClosedLedgerNum() >
+                       lcl + 2;
+            },
+            3 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+        for (auto const& node : simulation->getNodes())
+        {
+            // Applied txs were removed and banned
+            REQUIRE(node->getHerder().getTx(tx1->getFullHash()) == nullptr);
+            REQUIRE(node->getHerder().getTx(tx2->getFullHash()) == nullptr);
+            REQUIRE(node->getHerder().isBannedTx(tx1->getFullHash()));
+            REQUIRE(node->getHerder().isBannedTx(tx2->getFullHash()));
+            // Both accounts are in the ledger
+            LedgerTxn ltx(node->getLedgerTxnRoot());
+            REQUIRE(stellar::loadAccount(ltx, a1.getPublicKey()));
+            REQUIRE(stellar::loadAccount(ltx, b1.getPublicKey()));
+        }
+    }
+    SECTION("all limited")
+    {
+        setup(false);
+
+        auto [root, a1, b1, tx1, tx2] = makeTxs(app);
+
+        // Submit txs for the same account, should be good
+        REQUIRE(app->getHerder().recvTransaction(tx1, true) ==
+                TransactionQueue::AddResult::ADD_STATUS_PENDING);
+
+        // Second tx is rejected due to limit
+        REQUIRE(app->getHerder().recvTransaction(tx2, true) ==
+                TransactionQueue::AddResult::ADD_STATUS_TRY_AGAIN_LATER);
+
+        uint32_t lcl = app->getLedgerManager().getLastClosedLedgerNum();
+        simulation->crankUntil(
+            [&]() {
+                return app->getLedgerManager().getLastClosedLedgerNum() >
+                       lcl + 2;
+            },
+            3 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
+
+        for (auto const& node : simulation->getNodes())
+        {
+            // Applied txs were removed and banned
+            REQUIRE(node->getHerder().getTx(tx1->getFullHash()) == nullptr);
+            REQUIRE(node->getHerder().getTx(tx2->getFullHash()) == nullptr);
+            REQUIRE(node->getHerder().isBannedTx(tx1->getFullHash()));
+            // Second tx is not banned because it's never been flooded and
```

Comment:
I think it would make sense to also submit and get this tx applied to make sure we don't have any staleness issues for the 'prohibited' accounts set.

Reply:
Good idea, added.